pkg/cmd/certregenerationcontroller/cmd.go

```diff
@@ -0,0 +1,169 @@
+	configexternalinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/certrotationcontroller"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/version"
+)
+
+type Options struct {
+	controllerContext *controllercmd.ControllerContext
+
+	TLSServerName string
+}
+
+func NewCertRegenerationControllerCommand(ctx context.Context) *cobra.Command {
+	o := &Options{
+		TLSServerName: "localhost-recovery",
+	}
+
+	cmd := controllercmd.
+		NewControllerCommandConfig("cert-regeneration-controller", version.Get(), func(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+			o.controllerContext = controllerContext
+
+			err := o.Validate(ctx)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(ctx)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(ctx)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}).NewCommandWithContext(ctx)
+	cmd.Use = "cert-regeneration-controller"
+	cmd.Short = "Start the Cluster Certificate Regeneration Controller"
+
+	cmd.PersistentFlags().StringVarP(&o.TLSServerName, "tls-server-name", "", o.TLSServerName, "The SNI hostname to set for the server in kubeconfig")
+
+	return cmd
+}
+
+func (o *Options) Validate(ctx context.Context) error {
+	return nil
+}
+
+func (o *Options) Complete(ctx context.Context) error {
+	return nil
+}
+
+func (o *Options) Run(ctx context.Context) error {
+	if len(o.TLSServerName) != 0 {
+		// TLSServerName chooses the SNI serving endpoint on the apiserver.
+		// Particularly useful when connecting to "localhost" and wanting to choose a special
+		// serving endpoint like "localhost-recovery" that has long-lived serving certs
+		// for localhost connections.
+		o.controllerContext.KubeConfig.TLSClientConfig.ServerName = o.TLSServerName
+		o.controllerContext.ProtoKubeConfig.TLSClientConfig.ServerName = o.TLSServerName
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(o.controllerContext.ProtoKubeConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes client: %w", err)
+	}
+
+	configClient, err := configeversionedclient.NewForConfig(o.controllerContext.KubeConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create config client: %w", err)
+	}
+
+	configInformers := configexternalinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+
+	kubeAPIServerInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		kubeClient,
+		metav1.NamespaceAll,
```

Comment:
why this one.  Do you have cluster scoped resources?

Reply:
https://github.com/openshift/cluster-kube-apiserver-operator/pull/635#discussion_r369546475